services/graphql-server/src/graphql/resolvers/email/schedule.js

```diff
@@ -15,6 +17,57 @@ const clearSeconds = (date) => {
 };
 
 module.exports = {
+  /**
+   *
+   */
+  Query: {
+
+    emailSchedulesForDateRange: async (_, { input }, { basedb, site }) => {
+      const {
+        newsletterId,
+        beforeEndOf,
+        startingStartOf,
+      } = input;
+
+      // Use input timezone otherwise fallback to site/global timezone.
+      const timezone = input.timezone || site.get('date.timezone', defaults.date.timezone);
+      const startDate = momentTZ(startingStartOf).tz(timezone);
+      const endDate = momentTZ(beforeEndOf).tz(timezone);
+      const start = startDate.startOf('day').toDate();
+      const end = endDate.endOf('day').toDate();
+
+      const scheduleSort = { sequence: 1, deploymentDate: 1 };
+
+      const scheduleQuery = {
+        product: BaseDB.coerceID(newsletterId),
+        status: 1,
+        deploymentDate: { $gte: start, $lte: end },
+      };
+      const schedules = await basedb.find('email.Schedule', scheduleQuery, {
+        sort: scheduleSort,
+        projection: {
+          _id: 1,
+          product: 1,
+          'content.$id': 1,
+          section: 1,
+          deploymentDate: 1,
+          sequence: 1,
+          status: 1,
+        },
+      });
+      const contentIds = schedules.map((schedule) => BaseDB.extractRefId(schedule.content));
+
+      if (!contentIds.length) return [];
+      const output = schedules.reduce((array, schedule) => {
+        const { content: contentItem, ...rest } = schedule;
+        if (contentItem && contentItem.oid) {
+          array.push({ ...rest, content: contentItem.oid });
+        }
+        return array;
+      }, []);
+      return output;
```

Comment:
What are we doing here?

There should be no active schedules that do not have active content. The shape of the returned `EmailSchedule` should not be modified. If you _have_ to make the content reference only the id, you need to create and return a new type. But the better approach is to just return the id field when querying.

You should just be able to use `return basedb.find(...)` here.

Reply:
Whether or not it should be the case, "in the real world" there are schedules that do not have active content, otherwise the `contentStatus` field would always be 1 and to that point there wouldn't really be a reason to have it, to that point it would appear that even when a content item is set to draft (status 2) the `contentStatus` on it's respective email schedule still indicates 1 (published) which is inherently not correct.

Here are some "real world examples":
![Screenshot from 2024-03-15 11-58-59](https://github.com/parameter1/base-cms/assets/46794001/be37212d-7cc6-40d7-b3d1-468ea5bf7cff)
![Screenshot from 2024-03-15 11-59-10](https://github.com/parameter1/base-cms/assets/46794001/db6f83c1-46b7-4eec-a7d7-a86f0b8a265e)


Lastly an example of `contentStatus` functioning incorrectly:
https://manage.parameter1.dev/content/edit/article/22879537
![Screenshot from 2024-03-15 12-02-52](https://github.com/parameter1/base-cms/assets/46794001/51444c8d-0cee-47de-a459-9aa1798092c4)
![Screenshot from 2024-03-15 12-02-59](https://github.com/parameter1/base-cms/assets/46794001/d7e3a1d7-a020-401f-b6bb-5640c543972a)

To directly answer the question this is actually done to get the resolution to work correctly as this winds up using the following: https://github.com/parameter1/base-cms/blob/master/services/graphql-server/src/dataloaders/utils/run-queries.js#L72 which if I had returned this result "raw" without errors due to the values being passed in as `ids` there being objects over numerical values. What was done here was simply to correct the resolution as I believe it's attempting to use this: https://github.com/parameter1/base-cms/blob/master/services/graphql-server/src/graphql/definitions/email/schedule.js#L30 which seemingly requires it to be an id reference in order to perform something akin to `basedb.find(...)` inherently.

Reply:
Ok, I think i followed all of that. You still should not need to modify the values.

The `EmailSchedule.content` field uses the `refOne` directive, which uses the `extractRefId` utility function that will handle the conversion to a raw id from the dbref format: https://github.com/parameter1/base-cms/blob/master/services/graphql-server/src/graphql/directives/ref-one.js#L34

That may use the dataloader/query aggregator under the hood, but you don't need to change the reference format in order for it to work. What result are you seeing that makes you think it won't work correctly?

Compare it to the `contentEmailSchedules` query, which returns this type. The content field works as expected there, so what's different here?

<img width="743" alt="image" src="https://github.com/parameter1/base-cms/assets/1778222/21cad961-4129-4839-b543-da9619dcace5">

Reply:
It looks like this might have been an artifact of however I was resolving this before, potentially due to something missing in my projection, I believe this to be resolved and I have removed this unnecessary rebuilding via: https://github.com/parameter1/base-cms/pull/882/commits/48bdeb75dcb080be870cd21f8c62d3015a7924b7